### ISSUE TITLE
Prometheus Custom metrics not working as expected

### ISSUE THREAD

**Issue by brahama** [TykTechnologies/tyk-pump#512]
Hi fellows,

I managed to make Prometheus pump work after a few modifications to the Helms chart.
But now i have a problem with Custom metrics. Looks like its only applying the last one form the list. I tested it adding a few and i only get metrics for the last one.

Adding some logs that might point to the problem as you can see it process 3 times the same one...

```
time="Nov  6 01:14:01" level=debug msg="Fixed keyname is: analytics-tyk-system-analytics"
time="Nov  6 01:14:01" level=debug msg="Unpacked vals: 1"
time="Nov  6 01:14:01" level=debug msg="Decoded Record: {GET echo.tyk-demo.com:8080 / / 0 curl/7.79.1 6 November 2022 1 200 00000000 2022-11-06 01:14:00.257824001 +0000 UTC Non Versioned Hello-World Hello-World 1  179   172.31.0.1 {{} {0 map[]} {0 0 }} {0 0 0 0} {179 178} [key-00000000 org-1 api-Hello-World]  false 2122-10-13 01:14:00.257833966 +0000 UTC}"
time="Nov  6 01:14:01" level=debug msg="Writing to: Prometheus Pump"
time="Nov  6 01:14:01" level=debug msg="Attempting to write 1 records..."
time="Nov  6 01:14:01" level=debug msg="Processing metric:tyk_http_requests_total"
time="Nov  6 01:14:01" level=debug msg="Processing metric:tyk_http_requests_total"
time="Nov  6 01:14:01" level=debug msg="Processing metric:tyk_http_requests_total"
time="Nov  6 01:14:01" level=info msg="Purged 1 records..."
time="Nov  6 01:14:01" level=debug msg="Getting raw key set: tyk-system-analytics_protobuf"
```

And my pump.conf is the next one and i have 3 custom metrics

```
{
  "log_level": "info",
  "analytics_storage_type": "redis",
  "analytics_storage_config": {
    "type": "redis",
    "hosts": null,
    "username": "",
    "optimisation_max_idle": 2000,
    "optimisation_max_active": 4000
  },
  "pumps": {
    "prometheus": {
      "type": "prometheus",
      "meta": {
        "listen_address": "0.0.0.0:9090",
        "path": "/metrics",
        "custom_metrics":[
          {
            "name":"tyk_custom_http_status_per_api_name",
            "description":"This is a custom counter",
            "metric_type":"counter",
            "labels":["response_code","api_name"]
          },
          {
            "name":"tyk_http_latency",
            "description":"Latency of API requests",
            "metric_type":"histogram",
            "labels":["response_code","api_name","method","api_key","path"]
          },
          {
            "name":"tyk_http_requests_total",
            "description":"Total of API requests",
            "metric_type":"counter",
            "labels":["response_code","api_name","method","api_key","path"]
          }
        ]
      }
    }
  },
  "dont_purge_uptime_data": true,
  "purge_delay": 2
}
```

I have no env variables for this.

Thanks!

**Comment by brahama** [comment]
Indeed. You need to point to master/or 1.7 or 1.6.1-rc4

**Comment by tbuchaillot** [comment]
Hey @brahama ! This is already solved on the next Pump release - v1.7.0.  Thanks for raising the issue!
